apps/service_providers/models.py

```diff
@@ -164,6 +180,79 @@ def get_speech_service(self) -> speech_service.SpeechService:
         config = {k: v for k, v in self.config.items() if v}
         return self.type_enum.get_speech_service(config)
 
+    def validate_uploaded_file_names(self, uploaded_files: list[File]):
+        """Validate file types. Raises UserServiceProviderConfigError for invalid file types"""
+        if self.type == VoiceProviderType.openai_voice_engine:
+            accepted_file_types = ["mp4", "mp3"]
+            invalid_extentions = set()
+            for file in uploaded_files:
+                file_extention = file.name.split(".")[1]
+                if file_extention not in accepted_file_types:
+                    invalid_extentions.add(f".{file_extention}")
+            if invalid_extentions:
+                string = ", ".join(invalid_extentions)
+                raise UserServiceProviderConfigError(f"File extentions not supported: {string}")
```

Comment:
I think you could use a standard Django `ValidationError` here.

Also, what about doing this validation during form validation instead of on save? You could extend the `BaseFileFormSet` and update the `get_file_formset` to take in a form class.